canal/sync.go

```diff
@@ -175,6 +179,10 @@ func (c *Canal) runSyncBinlog() error {
 	return nil
 }
 
+func (c *Canal) updateReplicationDelay(ev *replication.BinlogEvent) {
+	c.delay = uint32(time.Now().Unix()) - ev.Header.Timestamp
```

Comment:
maybe we should use atomic to save delay

Reply:
👍🏻 Yeah somehow didnt think about it